src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -67,190 +68,186 @@ async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)>
-                    {
-                        AddNamesForTypeWorker(constraintType, builder);
-                    }
+                    indicesComplete = false;
+                    continue;
                 }
-                else
+
+                foreach (var symbol in result)
                 {
-                    builder.Add(GetReceiverTypeName(receiverTypeSymbol));
-                    builder.AddRange(receiverTypeSymbol.GetBaseTypes().Select(t => t.MetadataName));
-                    builder.AddRange(receiverTypeSymbol.GetAllInterfacesIncludingThis().Select(t => t.MetadataName));
+                    var containingNamespacename = GetFullyQualifiedNamespaceName(symbol.ContainingNamespace, namespaceNameCache);
 
-                    // interface doesn't inherit from object, but is implicitly convertible to object type.
-                    if (receiverTypeSymbol.IsInterfaceType())
+                    IMethodSymbol bestSymbol;
+                    int overloadCount;
+
+                    // Select the overload with minimum number of parameters to display
+                    if (overloadMap.TryGetValue((containingNamespacename, symbol.Name, symbol.Arity > 0), out var currentValue))
                     {
-                        builder.Add(nameof(Object));
+                        bestSymbol = currentValue.bestSymbol.Parameters.Length > symbol.Parameters.Length ? symbol : currentValue.bestSymbol;
+                        overloadCount = currentValue.overloadCount + 1;
                     }
-                }
-            }
-        }
-
-        private static string GetReceiverTypeName(ITypeSymbol typeSymbol)
-        {
-            switch (typeSymbol)
-            {
-                case INamedTypeSymbol namedType:
-                    return namedType.MetadataName;
-
-                case IArrayTypeSymbol arrayType:
-                    var elementType = arrayType.ElementType;
-                    while (elementType is IArrayTypeSymbol symbol)
+                    else
                     {
-                        elementType = symbol.ElementType;
+                        bestSymbol = symbol;
+                        overloadCount = 1;
                     }
 
-                    var elementTypeName = GetReceiverTypeName(elementType);
-
-                    // We do not differentiate array of different kinds sicne they are all represented in the indices as "NonArrayElementTypeName[]"
-                    // e.g. int[], int[][], int[,], etc. are all represented as "int[]", whereas array of complex type such as T[] is "[]".
-                    return elementTypeName + FindSymbols.Extensions.ArrayReceiverTypeNameSuffix;
-
-                default:
-                    // Complex types are represented by "";
-                    return FindSymbols.Extensions.ComplexReceiverTypeName;
+                    overloadMap[(containingNamespacename, symbol.Name, symbol.Arity > 0)] = (bestSymbol, overloadCount);
```

Comment:
this tuple is duplicated with above.  can you extract out to something like `var overloadKey = (...)`.  i think that would make this easier to read.